{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d552447c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-04 13:56:33.827673: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory\n",
      "2022-04-04 13:56:33.827700: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import torch\n",
    "\n",
    "from pytorch_widedeep import Trainer\n",
    "from pytorch_widedeep.preprocessing import (\n",
    "    WidePreprocessor,\n",
    "    TabPreprocessor,\n",
    "    TextPreprocessor,\n",
    "    ImagePreprocessor,\n",
    ")\n",
    "from pytorch_widedeep.models import (\n",
    "    Wide,\n",
    "    TabMlp,\n",
    "    Vision,\n",
    "    BasicRNN,\n",
    "    WideDeep,\n",
    ")\n",
    "from pytorch_widedeep.losses import RMSELoss\n",
    "from pytorch_widedeep.initializers import *\n",
    "from pytorch_widedeep.callbacks import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8cd6360e",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"./RatingswithUI.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c5633211",
   "metadata": {},
   "outputs": [],
   "source": [
    "ind = np.where([not isinstance(f, float) for f in df.feature])[0]\n",
    "df = df.iloc[ind, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8075f11d",
   "metadata": {},
   "outputs": [],
   "source": [
    "crossed_cols = [(\"verified\", \"tot_review_u>10\")]\n",
    "# add binary variable for tot_review\n",
    "wide_cols = [\"verified\", \"tot_review_u>10\"]\n",
    "\n",
    "cat_embed_cols = [(\"main_cat\", 29)]\n",
    "\n",
    "continuous_cols = [\"avg_score_u\", \"avg_sentiment_u\", \n",
    "                   \"weighted_score_u\", \"avg_score\", \n",
    "                   \"avg_sentiment\", \"weighted_score\",\n",
    "                  \"tot_review\"]\n",
    "\n",
    "text_col = \"feature\"\n",
    "img_col = \"img_col\"\n",
    "\n",
    "word_vectors_path = \"./glove.6B/glove.6B.100d.txt\"\n",
    "img_path = \"./data/Images for use\"\n",
    "\n",
    "target_col = \"overall\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2273d322",
   "metadata": {},
   "outputs": [],
   "source": [
    "target = df[target_col].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7dbc9dbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "wide_preprocessor = WidePreprocessor(wide_cols=wide_cols, crossed_cols=crossed_cols)\n",
    "X_wide = wide_preprocessor.fit_transform(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8538266b",
   "metadata": {},
   "outputs": [],
   "source": [
    "tab_preprocessor = TabPreprocessor(\n",
    "    cat_embed_cols=cat_embed_cols,\n",
    "    continuous_cols=continuous_cols,\n",
    ")\n",
    "X_tab = tab_preprocessor.fit_transform(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e967d3e0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c6489ee5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The vocabulary contains 1966 tokens\n",
      "Indexing word vectors...\n",
      "Loaded 400000 word vectors\n",
      "Preparing embeddings matrix...\n",
      "1761 words in the vocabulary had ./glove.6B/glove.6B.100d.txt vectors and appear more than 5 times\n"
     ]
    }
   ],
   "source": [
    "text_preprocessor = TextPreprocessor(\n",
    "    word_vectors_path=word_vectors_path, text_col=text_col\n",
    ")\n",
    "X_text = text_preprocessor.fit_transform(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "140427c5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading Images from ./data/Images for use\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  2%|▉                                          | 442/20000 [00:00<00:04, 4413.94it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Resizing\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 20000/20000 [00:07<00:00, 2564.69it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Computing normalisation metrics\n"
     ]
    }
   ],
   "source": [
    "image_processor = ImagePreprocessor(img_col=img_col, img_path=img_path)\n",
    "X_images = image_processor.fit_transform(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "65c3c7aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "wide = Wide(input_dim=np.unique(X_wide).shape[0], pred_dim=1)\n",
    "\n",
    "# DeepDense: 2 Dense layers\n",
    "tab_mlp = TabMlp(\n",
    "    column_idx=tab_preprocessor.column_idx,\n",
    "    cat_embed_input=tab_preprocessor.cat_embed_input,\n",
    "    cat_embed_dropout=0.1,\n",
    "    continuous_cols=continuous_cols,\n",
    "    mlp_hidden_dims=[128, 64],\n",
    "    mlp_dropout=0.1,\n",
    ")\n",
    "\n",
    "# DeepText: a stack of 2 LSTMs\n",
    "basic_rnn = BasicRNN(\n",
    "    vocab_size=len(text_preprocessor.vocab.itos),\n",
    "    embed_matrix=text_preprocessor.embedding_matrix,\n",
    "    n_layers=4,\n",
    "    hidden_dim=64,\n",
    "    rnn_dropout=0.5,\n",
    ")\n",
    "\n",
    "# Pretrained Resnet 18\n",
    "resnet = Vision(pretrained_model_name=\"resnet34\", n_trainable=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "826c8bb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "model = WideDeep(\n",
    "    wide=wide,\n",
    "    deeptabular=tab_mlp,\n",
    "    deeptext=basic_rnn,\n",
    "    deepimage=resnet,\n",
    "    head_hidden_dims=[256, 128],\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d54e504f",
   "metadata": {},
   "outputs": [],
   "source": [
    "deep_params = []\n",
    "for childname, child in model.named_children():\n",
    "    if childname == \"deeptabular\":\n",
    "        for n, p in child.named_parameters():\n",
    "            if \"embed_layer\" in n:\n",
    "                deep_params.append({\"params\": p, \"lr\": 1e-4})\n",
    "            else:\n",
    "                deep_params.append({\"params\": p, \"lr\": 1e-3})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7c4c5c97",
   "metadata": {},
   "outputs": [],
   "source": [
    "wide_opt = torch.optim.Adam(model.wide.parameters(), lr=0.03)\n",
    "deep_opt = torch.optim.Adam(deep_params)\n",
    "text_opt = torch.optim.AdamW(model.deeptext.parameters())\n",
    "img_opt = torch.optim.AdamW(model.deepimage.parameters())\n",
    "head_opt = torch.optim.Adam(model.deephead.parameters())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9c29b0a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "wide_sch = torch.optim.lr_scheduler.StepLR(wide_opt, step_size=5)\n",
    "deep_sch = torch.optim.lr_scheduler.MultiStepLR(deep_opt, milestones=[3, 8])\n",
    "text_sch = torch.optim.lr_scheduler.StepLR(text_opt, step_size=5)\n",
    "img_sch = torch.optim.lr_scheduler.MultiStepLR(deep_opt, milestones=[3, 8])\n",
    "head_sch = torch.optim.lr_scheduler.StepLR(head_opt, step_size=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "39c264fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "optimizers = {\n",
    "    \"wide\": wide_opt,\n",
    "    \"deeptabular\": deep_opt,\n",
    "    \"deeptext\": text_opt,\n",
    "    \"deepimage\": img_opt,\n",
    "    \"deephead\": head_opt,\n",
    "}\n",
    "schedulers = {\n",
    "    \"wide\": wide_sch,\n",
    "    \"deeptabular\": deep_sch,\n",
    "    \"deeptext\": text_sch,\n",
    "    \"deepimage\": img_sch,\n",
    "    \"deephead\": head_sch,\n",
    "}\n",
    "\n",
    "# Now...we have used pretrained word embeddings, so you do not want to\n",
    "# initialise these  embeddings. However you might still want to initialise the\n",
    "# other layers in the DeepText component. No probs, you can do that with the\n",
    "# parameter pattern and your knowledge on regular  expressions. Here we are\n",
    "# telling to the KaimingNormal initializer to NOT touch the  parameters whose\n",
    "# name contains the string word_embed.\n",
    "initializers = {\n",
    "    \"wide\": KaimingNormal,\n",
    "    \"deeptabular\": KaimingNormal,\n",
    "    \"deeptext\": KaimingNormal(pattern=r\"^(?!.*word_embed).*$\"),\n",
    "    \"deepimage\": KaimingNormal,\n",
    "}\n",
    "\n",
    "mean = [0.406, 0.456, 0.485]  # BGR\n",
    "std = [0.225, 0.224, 0.229]  # BGR\n",
    "transforms = [ToTensor, Normalize(mean=mean, std=std)]\n",
    "callbacks = [\n",
    "    LRHistory(n_epochs=10),\n",
    "    EarlyStopping,\n",
    "    ModelCheckpoint(filepath=\"model_weights/wd_out\"),\n",
    "]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ba08db5a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/initializers.py:34: UserWarning: No initializer found for deephead\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "trainer = Trainer(\n",
    "    model,\n",
    "    objective=\"rmse\",\n",
    "    initializers=initializers,\n",
    "    optimizers=optimizers,\n",
    "    lr_schedulers=schedulers,\n",
    "    callbacks=callbacks,\n",
    "    transforms=transforms,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4b0fde41",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:   0%|                                                | 0/500 [00:06<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /opt/conda/conda-bld/pytorch_1623448234945/work/c10/core/TensorImpl.h:1156.)\n",
      "  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)\n",
      "epoch 1:   0%|                             | 1/500 [00:06<54:08,  6.51s/it, loss=4.12]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  32%|████████▉                   | 159/500 [00:15<00:19, 17.17it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  36%|██████████▏                 | 181/500 [00:17<00:18, 16.98it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  40%|███████████▏                | 199/500 [00:18<00:18, 16.69it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  41%|███████████▌                | 207/500 [00:18<00:17, 16.78it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  51%|██████████████▏             | 253/500 [00:21<00:15, 16.37it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  51%|██████████████▎             | 255/500 [00:21<00:15, 16.29it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  53%|██████████████▋             | 263/500 [00:22<00:14, 16.63it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1:  66%|██████████████████▌         | 331/500 [00:26<00:10, 15.76it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 1: 100%|████████████████████████████| 500/500 [00:36<00:00, 13.76it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid:   0%|                                                  | 0/125 [00:06<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid:   1%|▎                               | 1/125 [00:06<12:53,  6.23s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:10<00:00, 12.35it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:   0%|                                                | 0/500 [00:06<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:   0%|                              | 1/500 [00:06<53:36,  6.45s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:   0%|                              | 1/500 [00:06<53:36,  6.45s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  32%|████████▉                   | 160/500 [00:15<00:21, 16.13it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  36%|██████████                  | 180/500 [00:16<00:19, 16.10it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  40%|███████████                 | 198/500 [00:17<00:17, 17.51it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  42%|███████████▋                | 208/500 [00:18<00:17, 16.47it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  50%|██████████████              | 252/500 [00:21<00:15, 16.24it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  51%|██████████████▎             | 256/500 [00:21<00:14, 16.44it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  52%|██████████████▋             | 262/500 [00:21<00:14, 16.17it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2:  66%|██████████████████▌         | 332/500 [00:26<00:10, 15.61it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 2: 100%|████████████████████████████| 500/500 [00:36<00:00, 13.67it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:10<00:00, 11.73it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:   0%|                              | 1/500 [00:06<56:47,  6.83s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  32%|████████▉                   | 159/500 [00:16<00:20, 16.37it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  36%|██████████▏                 | 181/500 [00:17<00:18, 17.47it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  40%|███████████▏                | 199/500 [00:18<00:17, 17.26it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  41%|███████████▌                | 207/500 [00:18<00:17, 17.10it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  51%|██████████████▏             | 253/500 [00:21<00:15, 16.41it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  51%|██████████████▎             | 255/500 [00:21<00:14, 16.55it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  53%|██████████████▋             | 263/500 [00:22<00:14, 16.80it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3:  66%|██████████████████▌         | 331/500 [00:26<00:10, 16.56it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 3: 100%|████████████████████████████| 500/500 [00:36<00:00, 13.81it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:11<00:00, 10.61it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:   0%|                            | 1/500 [00:08<1:07:50,  8.16s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  32%|████████▉                   | 159/500 [00:16<00:17, 19.34it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  36%|██████████                  | 180/500 [00:17<00:18, 17.40it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  40%|███████████▏                | 199/500 [00:19<00:18, 16.41it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  41%|███████████▌                | 207/500 [00:19<00:17, 16.71it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  50%|██████████████              | 252/500 [00:22<00:15, 16.52it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  51%|██████████████▎             | 256/500 [00:22<00:14, 16.36it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  52%|██████████████▋             | 262/500 [00:22<00:14, 16.96it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4:  66%|██████████████████▌         | 332/500 [00:26<00:09, 16.84it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 4: 100%|████████████████████████████| 500/500 [00:36<00:00, 13.64it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:11<00:00, 10.65it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:   0%|                            | 1/500 [00:08<1:08:21,  8.22s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  32%|████████▉                   | 159/500 [00:16<00:17, 19.65it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  36%|██████████                  | 180/500 [00:17<00:18, 17.42it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  40%|███████████                 | 198/500 [00:18<00:18, 15.95it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  42%|███████████▋                | 208/500 [00:18<00:18, 15.56it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  51%|██████████████▏             | 253/500 [00:21<00:12, 20.07it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  51%|██████████████▎             | 256/500 [00:21<00:12, 19.95it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  52%|██████████████▋             | 262/500 [00:21<00:12, 19.51it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5:  66%|██████████████████▌         | 331/500 [00:25<00:08, 19.73it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 5: 100%|████████████████████████████| 500/500 [00:33<00:00, 14.73it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:11<00:00, 10.44it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:   0%|                            | 1/500 [00:07<1:01:18,  7.37s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:   0%|                            | 1/500 [00:07<1:01:18,  7.37s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:   1%|▏                             | 3/500 [00:07<16:14,  1.96s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  32%|████████▉                   | 159/500 [00:16<00:19, 17.63it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  36%|██████████▏                 | 181/500 [00:18<00:20, 15.33it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  40%|███████████▏                | 199/500 [00:19<00:17, 17.11it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  41%|███████████▌                | 207/500 [00:19<00:17, 16.43it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  51%|██████████████▏             | 253/500 [00:22<00:15, 16.02it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  51%|██████████████▎             | 255/500 [00:22<00:15, 15.99it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  53%|██████████████▋             | 263/500 [00:23<00:14, 16.08it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6:  66%|██████████████████▌         | 331/500 [00:27<00:10, 16.86it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 6: 100%|████████████████████████████| 500/500 [00:37<00:00, 13.26it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:10<00:00, 11.88it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:   0%|                                                | 0/500 [00:06<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:   0%|                              | 1/500 [00:06<52:47,  6.35s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:   0%|                              | 1/500 [00:06<52:47,  6.35s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  32%|████████▉                   | 159/500 [00:15<00:18, 18.76it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  36%|██████████▏                 | 181/500 [00:17<00:19, 16.59it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  40%|███████████▏                | 199/500 [00:18<00:19, 15.70it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  41%|███████████▌                | 207/500 [00:18<00:19, 15.40it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  51%|██████████████▏             | 253/500 [00:21<00:14, 16.99it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  51%|██████████████▎             | 255/500 [00:21<00:14, 16.56it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  53%|██████████████▋             | 263/500 [00:22<00:13, 17.24it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7:  66%|██████████████████▌         | 331/500 [00:26<00:10, 16.24it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 7: 100%|████████████████████████████| 500/500 [00:36<00:00, 13.72it/s, loss=nan]\n",
      "valid:   0%|                                                  | 0/125 [00:07<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid:   1%|▎                               | 1/125 [00:07<14:34,  7.05s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid:   1%|▎                               | 1/125 [00:07<14:34,  7.05s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid:   1%|▎                               | 1/125 [00:07<14:34,  7.05s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "valid: 100%|██████████████████████████████| 125/125 [00:11<00:00, 11.22it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:   0%|                              | 1/500 [00:06<56:19,  6.77s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  32%|████████▉                   | 159/500 [00:15<00:17, 19.91it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  36%|██████████                  | 180/500 [00:16<00:16, 19.00it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  40%|███████████                 | 198/500 [00:17<00:17, 16.82it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  41%|███████████▌                | 207/500 [00:18<00:15, 18.71it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  51%|██████████████▏             | 253/500 [00:21<00:15, 16.45it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  51%|██████████████▎             | 255/500 [00:21<00:14, 16.70it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  53%|██████████████▋             | 263/500 [00:21<00:14, 16.67it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8:  66%|██████████████████▌         | 331/500 [00:25<00:10, 16.80it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 8: 100%|████████████████████████████| 500/500 [00:35<00:00, 14.12it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:11<00:00, 10.68it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:   0%|                            | 1/500 [00:08<1:08:50,  8.28s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:  32%|████████▉                   | 160/500 [00:17<00:18, 18.78it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:  36%|██████████▏                 | 181/500 [00:18<00:19, 16.74it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "epoch 9:  40%|███████████▏                | 199/500 [00:19<00:17, 17.26it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:  41%|███████████▌                | 207/500 [00:19<00:17, 16.80it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:  50%|██████████████              | 251/500 [00:22<00:13, 17.88it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:  51%|██████████████▎             | 256/500 [00:22<00:13, 18.31it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:  52%|██████████████▋             | 262/500 [00:22<00:13, 17.52it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9:  66%|██████████████████▌         | 332/500 [00:26<00:10, 16.17it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 9: 100%|████████████████████████████| 500/500 [00:36<00:00, 13.55it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:12<00:00, 10.37it/s, loss=nan]\n",
      "  0%|                                                         | 0/500 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:   0%|                                               | 0/500 [00:08<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:   0%|                           | 1/500 [00:08<1:08:34,  8.25s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:   0%|                           | 1/500 [00:08<1:08:34,  8.25s/it, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  32%|████████▌                  | 159/500 [00:17<00:20, 16.84it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  36%|█████████▊                 | 181/500 [00:18<00:16, 19.27it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  40%|██████████▋                | 198/500 [00:19<00:15, 19.82it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  42%|███████████▏               | 208/500 [00:19<00:15, 19.06it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  50%|█████████████▌             | 252/500 [00:22<00:15, 16.46it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  51%|█████████████▊             | 256/500 [00:22<00:14, 16.64it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  52%|██████████████▏            | 262/500 [00:22<00:14, 16.85it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10:  66%|█████████████████▉         | 332/500 [00:26<00:10, 16.79it/s, loss=nan]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "epoch 10: 100%|███████████████████████████| 500/500 [00:37<00:00, 13.48it/s, loss=nan]\n",
      "  0%|                                                         | 0/125 [00:00<?, ?it/s]/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "/home/peilin/anaconda3/lib/python3.8/site-packages/pytorch_widedeep/training/_wd_dataset.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  xdi = (xdi / xdi.max()).astype(\"float32\")\n",
      "valid: 100%|██████████████████████████████| 125/125 [00:11<00:00, 11.04it/s, loss=nan]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model weights after training corresponds to the those of the final epoch which might not be the best performing weights. Usethe 'ModelCheckpoint' Callback to restore the best epoch weights.\n"
     ]
    }
   ],
   "source": [
    "trainer.fit(\n",
    "    X_wide=X_wide,\n",
    "    X_tab=X_tab,\n",
    "    X_text=X_text,\n",
    "    X_img=X_images,\n",
    "    target=target,\n",
    "    n_epochs=1000,\n",
    "    batch_size=32,\n",
    "    val_split=0.2,\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
